FaunaDB.Client/Query/Language.Authentication.cs

```diff
@@ -45,19 +45,91 @@ public partial struct Language
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/currentidentity">docs</see>.
+        /// </para>
+        /// </summary>
+        public static Expr CurrentIdentity() =>
+            UnescapedObject.With("current_identity", Null());
 
         /// <summary>
         /// Creates a new HasIdentity expression.
         /// <para>
-        /// See the <see href="https://fauna.com/documentation/queries#auth_functions">docs</see>.
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/hasidentity">docs</see>.
         /// </para>
         /// </summary>
         public static Expr HasIdentity() =>
             UnescapedObject.With("has_identity", Null());
+        
+        /// <summary>
+        /// Creates a new HasCurrentIdentity expression.
+        /// <para>
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/hascurrentidentity">docs</see>.
+        /// </para>
+        /// </summary>
+        public static Expr HasCurrentIdentity() =>
+            UnescapedObject.With("has_current_identity", Null());
+        
+        /// <summary>
+        /// Creates a new CurrentToken expression.
+        /// <para>
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/currenttoken">docs</see>.
+        /// </para>
+        /// </summary>
+        public static Expr CurrentToken() =>
+            UnescapedObject.With("current_token", Null());
+        
+        /// <summary>
+        /// Creates a new HasCurrentToken expression.
+        /// <para>
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/hascurrenttoken">docs</see>.
+        /// </para>
+        /// </summary>
+        public static Expr HasCurrentToken() =>
+            UnescapedObject.With("has_current_token", Null());
+        
+        /// <summary>
+        /// Creates a new AccessProviders expression.
+        /// <para>
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/accessprovider">docs</see>.
+        /// </para>
+        /// </summary>
+        public static Expr AccessProviders(Expr scope = null) =>
+            UnescapedObject.With("access_providers", scope ?? Null());
+        
+        /// <summary>
+        /// Creates a new AccessProvider expression.
+        /// <para>
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/accessprovider">docs</see>.
+        /// </para>
+        /// </summary>
+        public static Expr AccessProvider(Expr name) =>
+            UnescapedObject.With("access_provider", name);
+        
+        /// <summary>
+        /// Creates a new AccessProvider expression.
+        /// <para>
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/accessprovider">docs</see>.
+        /// </para>
+        /// </summary>
+        public static Expr AccessProvider(Expr name, Expr scope) =>
```

Comment:
```suggestion
        public static Expr AccessProvider(Expr name, Expr scope = null) =>
```
and you can remove the `public static Expr AccessProvider(Expr name)` version

Reply:
removed this method, and added `null` as you suggested